indy_common/test/auth/metadata/helper.py

```diff
@@ -39,61 +43,66 @@ def set_auth_constraint(validator, auth_constraint):
     validator.auth_cons_strategy.get_auth_constraint = lambda a: auth_constraint
 
 
-def build_req_and_action(signatures, need_to_be_owner, amount=None):
+def build_req_and_action(action: Action):
     sig = None
     sigs = None
-    identifier = None
-
-    if signatures:
-        role = next(iter(signatures.keys()))
-        identifier = IDENTIFIERS[role][0]
+    identifier = IDENTIFIERS[action.author][0]
+    endorser_did = None
 
-    if len(signatures) == 1 and next(iter(signatures.values())) == 1:
+    # if there is only 1 sig from the Author - use `signature` instead of `signatures`
+    if len(action.sigs) == 1 and next(iter(action.sigs.values())) == 1 and next(
+            iter(action.sigs.keys())) == action.author:
         sig = 'signature'
     else:
-        sigs = {IDENTIFIERS[role][i]: 'signature' for role, sig_count in signatures.items() for i in range(sig_count)}
+        sigs = {IDENTIFIERS[role][i]: 'signature' for role, sig_count in action.sigs.items() for i in
+                range(sig_count)}
+
+    if action.endorser is not None:
+        endorser_did = IDENTIFIERS[action.endorser][0]
 
     operation = randomOperation()
-    if amount is not None:
-        operation[PLUGIN_FIELD] = amount
+    if action.amount is not None:
+        operation[PLUGIN_FIELD] = action.amount
 
     req = Request(identifier=identifier,
                   operation=operation,
                   signature=sig,
-                  signatures=sigs)
+                  signatures=sigs,
+                  endorser=endorser_did)
     action = AuthActionAdd(txn_type=req.operation[TYPE],
                            field='some_field',
                            value='new_value',
-                           is_owner=need_to_be_owner)
+                           is_owner=action.is_owner)
 
     return req, [action]
 
 
 def validate(auth_constraint,
```

Comment:
I think adding docstrings to `validate`, `is_expected` and `build_req_and_action` could help a lot with tests readability